src/include/buffer.h

```diff
@@ -54,9 +56,12 @@ struct buffer_set_args
     struct buffer_set   *bsa_set;
     size_t               bsa_nbufs;
     size_t               bsa_buf_size;
+    void                *bsa_region;
+    size_t               bsa_region_size;
     void                *bsa_alt_source;
     size_t               bsa_alt_source_size;
-    size_t               bsa_alt_source_used;
+    uint8_t              bsa_use_alt_source_buf;
+    size_t               bsa_used_off;
```

Comment:
Wouldn't a clean option be to add an alignment member to this structure which can be used to align the `bsa_alt_source` when provided?  It could be used in conjunction w/ a new flag `BUFSET_OPT_ALT_SOURCE_BUF_ALIGN`  that would cause the alignment member, `bsa_alt_source_align` to be considered.